crates/miden-proving-service-client/src/proving_service/batch_prover.rs

```diff
@@ -11,6 +6,10 @@ use tokio::sync::Mutex;
 
 use super::generated::api_client::ApiClient;
 use crate::{
+    alloc::{
+        string::{String, ToString},
+        sync::Arc,
+    },
```

Comment:
I would keep `alloc` imports as they were as these are coming from the Rust standard library rather than something we've defined (I'd treat all other similar imports like this too).